Automattic/jetpack
# JS Components: re-write Text component with TS

#### Changes proposed in this Pull Request:

* This PR converts `Text` component to TypeScript and cleans up the propTypes.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the stories of `Text` component work as expected
* Check the consumers of `Text` and related components and confirm that they work as expected
* Check for any regression


https://user-images.githubusercontent.com/18226415/166904283-2ce5acd8-fcfc-4c22-81f3-1f16ecb1b5c6.mov
